stdlib/REPL/src/REPL.jl

```diff
@@ -691,7 +691,7 @@ function return_callback(s)
 end
 
 find_hist_file() = get(ENV, "JULIA_HISTORY",
-    joinpath(homedir(), ".julia", "logs", "repl_history.jl"))
+    !isempty(DEPOT_PATH) ? joinpath(DEPOT_PATH[1], "logs", "repl_history.jl") : nothing)
```

Comment:
Actually, on second though, maybe its better to throw from this function instead. How about:
```
find_hist_file() = get(ENV, "JULIA_HISTORY",
    !isempty(DEPOT_PATH) ? joinpath(DEPOT_PATH[1], "logs", "repl_history.jl") :
    error("DEPOT_PATH is empty and ENV[\"JULIA_HISTORY\"] not set.")
```
This error will be printed in the try-catch block and is perhaps more informative, thoughts?